Dockerfile

```diff
@@ -0,0 +1,13 @@
+FROM node:20.18.0-slim AS build
+WORKDIR /app
+COPY . ./
+RUN npm install
+RUN npx nx reset
+RUN npm ci
+RUN npm run build
+
+# production environment
+FROM nginxinc/nginx-unprivileged::1.27-alpine3.20

```

Comment:
I think there is a ":" too many here. When i run the build command it fails at once, but when I remove one of the ":" it runs fine.

Reply:
I'm likely doing something wrong here, but I also can't seem to access the site when running the container locally.

Reply:
Good catch.I have pushed a change where I have removed the extra colon.

Could it be that you do not map the exported port to a local port on your machine?
I built the imagage with 

`docker build -t pxweb2:test .`

and ran it with 
`docker run --rm -p 8080:8080 pxweb2:test`

you could change the first `8080` port to something else if you already have something running on that port already.

Reply:
I ran 
` docker build . -t pxweb2/web`
and
`docker run -dp 8080:80 pxweb2/web`

I can't get access from the browser, but the command `docker exec -it musing_borg  curl localhost:8080` gave a response.

Reply:
I see you are mapping 8080 on your machine to container port 80. But the container is exposing the 8080 port (we are using the nginx unprivileged that uses port 8080 instead of 80 which require root access). 

That is also why curl works when you exec into the container. Because your hit the 8080 port where nginx is listening and serving the files.

Reply:
Thanks, that got it working. I was mixing up which side did what on the port mapping. Changed it and got it working. 

I did notice that refreshing the page, or manually changing the URL to something other than the base `localhost:8080/` gives me a 404 error.

Reply:
I think the way to solve this is to use the `try_files` directive in nginx https://www.frontendundefined.com/posts/tutorials/nginx-react-router-404/

Reply:
I have a working version, shall i commit it here?

Reply:
Yes, do it

Reply:
It solved the problem for me @SjurSutterudSagen could you also verify it?

Reply:
Works fine here now